### DISCUSSION TITLE
Actions: postgres connect from docker container

### DISCUSSION THREAD

**Discussion by dima-kov** [community/community#26675] [Actions]
<p>Could you please help me to define what’s wrong?<br>
I am able to connect to postgres service running python code directly in actions, but I can’t connect to postgers service from inside docker container, despite I run it with <code>network_mode=host</code>. Pls, see:</p>
<pre><code class="lang-auto">name: Development
on:
  push:
    branches:
      - '**'
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
        run: |
          printf "DATABASE_URL=postgres://savr_house:savr_house@localhost:5432/savr_house\n" &gt;&gt; .env.prod

      - name: Build
        run: docker build -t savr .

      - name: Test
        run: docker run --rm --network="host" savr python manage.py test

    services:
      postgres:
        image: postgres:14
        options: &gt;-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5
        env:
          POSTGRES_USER: savr_house
          POSTGRES_PASSWORD: savr_house
          POSTGRES_DB: savr_house
        ports:
          - 5432:5432

</code></pre>
<p>Could you help me please?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Three things:</p>
<ul>
<li>Your workflow is invalid at the point below, you can’s use <code>uses</code> and <code>run</code> in one step. Also, the <a href="https://github.com/actions/checkout"><code>actions/checkout</code></a> version is very outdated (v3 is current).</li>
</ul>
<aside class="quote no-group" data-username="dima-kov" data-post="1" data-topic="250152">
<div class="title">
<div class="quote-controls"></div>
 dima-kov:</div>
<blockquote>
<pre><code class="lang-yaml">      - uses: actions/checkout@v1
        run: |
          printf "DATABASE_URL=postgres://savr_house:savr_house@localhost:5432/savr_house\n" &gt;&gt; .env.prod
</code></pre>
</blockquote>
</aside>
<ul>
<li>
<p>How is the <code>DATABASE_URL</code> variable supposed to get to the running test container? I guess theoretically the <code>docker build</code> could copy the <code>.env.prod</code> file into the image, but that doesn’t really make sense for runtime configuration. If you want to set an environment variable for the program in the container you should do it in the <code>docker run</code> step.</p>
</li>
<li>
<p>There’s no need to use <code>--network=host</code>. Instead you should attach your test container to <a href="https://docs.github.com/en/actions/learn-github-actions/contexts#job-context">the network named in <code>job.container.network</code></a>, and then you can access the service container by it’s name (in your case <code>postgres</code>) instead of <code>localhost</code>.</p>
</li>
</ul>

**Comment by dima-kov** [comment]
<p>thanks a lot. 1 and 2 are more typo error during forming minimal code for Topic. Using <a href="https://docs.github.com/en/actions/learn-github-actions/contexts#job-context" rel="noopener nofollow ugc"> <code>job.container.network</code> </a> was really helpful!</p>
